src/Servers/Kestrel/Core/src/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -363,16 +377,12 @@ private ValueTask<FlushResult> FlushAsync(Http2Stream stream, bool writeHeaders,
                 return default;
             }
 
-            if (writeHeaders)
-            {
-                // write headers
-                WriteResponseHeadersUnsynchronized(stream.StreamId, stream.StatusCode, Http2HeadersFrameFlags.END_STREAM, (HttpResponseHeaders)stream.ResponseHeaders);
-            }
+            WriteResponseHeadersUnsynchronized(stream.StreamId, stream.StatusCode, Http2HeadersFrameFlags.END_STREAM, (HttpResponseHeaders)stream.ResponseHeaders);
```

Comment:
Does stream.ResponseHeaders always have headers here?

Reply:
I see what's going on. In other places, we write data and flush the final write. This path is special because we're ending with a flush of headers only.

The answer is yes, there will always be at least one response header (I asked as I mixed up ResponseTrailers with ResponseHeaders)